### DISCUSSION TITLE
Upgrade to 24.8.x with Object('json') tables. Upgrade from experimental object type to experimental json type.

### DISCUSSION THREAD

**Discussion by plewam** [ClickHouse/ClickHouse#70832] [General]
Hi All,

I currently have a ClickHouse instance running with several tables that contain `Object('json')` columns. I am afraid to do the upgrade to 24.8.x or 24.9.x. Will the experimental object type (json) continue to work even though a new `JSON` type was introduced in 24.8? In my first tests it looks like it continues to work.

Once upgraded I plan to migrate every `object(json)` table to the new `JSON` type.

Kind regards,

Mathis

**Comment by Algunenano** [comment]
Yes. It should continue to work. The code for Object('json') is still there and it should keep working as before (same features, same bugs :smile: )

**Reply by plewam** [reply]
@Algunenano Thank you very much for the prompt response. I upgraded my ClickHouse instance and everything still works fine. 

Do you possibly know an easy way to upgrade an existing table to the new `JSON` type? Currently I think of renaming my existing table, create a new one with the new `JSON` type and then migrate everything with a `select into`. Maybe there is a smarter way to do that?

**Reply by Algunenano** [reply]
Maybe @Avogar can mention something else, but I'd expect `Select into` to be the only way

**Reply by Avogar** [reply]
`ALTER` from `Object('json')` to new `JSON` is in the roadmap: https://github.com/ClickHouse/ClickHouse/issues/68428.

**Reply by plewam** [reply]
Hi @Avogar that sounds really nice, thank you!

**Reply by plewam** [reply]
@Avogar With the new 24.11 release I found this #71320 PR. Does this enable my above desired migration? Can I simply alter a table?
